pydantic/types.py

```diff
@@ -55,38 +56,30 @@ def validate(cls, v):
 
 
 class ConstrainedStr(str):
-    strip_whitespace: bool = None
-    min_length: int = None
-    max_length: int = None
-    curtail_length: int = None
-    regex = None
+    strip_whitespace = False
+    min_length: Optional[int] = None
+    max_length: Optional[int] = None
+    curtail_length: Optional[int] = None
+    regex: Optional[Pattern] = None
 
     @classmethod
     def get_validators(cls):
+        yield not_none_validator
         yield str_validator
+        yield anystr_strip_whitespace
+        yield anystr_length_validator
         yield cls.validate
 
     @classmethod
     def validate(cls, value: str) -> str:
-        if value is None:
-            raise TypeError('None is not an allow value')
-
-        if cls.strip_whitespace:
-            value = value.strip()
-
         v_len = len(value)
-        if cls.min_length is not None and v_len < cls.min_length:
-            raise ValueError(f'length less than minimum allowed: {cls.min_length}')
-
-        if cls.curtail_length:
-            if v_len > cls.curtail_length:
-                value = value[:cls.curtail_length]
-        elif cls.max_length is not None and v_len > cls.max_length:
-            raise ValueError(f'length greater than maximum allowed: {cls.max_length}')
+        if cls.curtail_length and v_len > cls.curtail_length:
```

Comment:
I guess here you can use

```py
if cls.curtail_length and len(value) > cls.curtail_length:
    ...
```

Since the `v_len` variable isn't reused.